src/plugins/gs-plugin-external-appstream.c

```diff
@@ -169,13 +170,22 @@ gs_plugin_external_appstream_refresh_url (GsPlugin *plugin,
 		return FALSE;
 
 	tmp_file_path = g_file_get_path (tmp_file);
+
 	g_debug ("Downloaded appstream file %s", tmp_file_path);
 
 	/* write to file */
 	outstream = g_io_stream_get_output_stream (G_IO_STREAM (iostream));
-	if (!g_output_stream_write_all (outstream, msg->response_body->data,
-					msg->response_body->length, NULL,
-					cancellable, error)) {
+	file_written = g_output_stream_write_all (outstream,
+						  msg->response_body->data,
+						  msg->response_body->length,
+						  NULL, cancellable, error);
+
+	/* close the file */
+	if (!g_output_stream_close (outstream, cancellable, error)) {
```

Comment:
You can't reuse `error` here, it may already have been set.
I would just always close the stream without error checking, not returning if it fails.